### PR TITLE
WOR-1419 use new listPao TPS api

### PR DESCRIPTION
**Pull request by dvoet**
https://broadworkbench.atlassian.net/browse/WOR-1419

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/app/controller/FolderApiControllerTest.java [hunk 1]
```diff
@@ -323,7 +323,8 @@ public void deleteFolders_deleteTopLevelFolder_folderAndSubFoldersAllDeleted() t
   private ApiJobReport deleteFolderAndWaitForJob(UUID workspaceId, ApiFolder folder)
       throws Exception {
     String serializedResponse =
-        deleteFolderExpectCode(workspaceId, folder.getId(), HttpStatus.SC_ACCEPTED)
+        deleteFolder(workspaceId, folder.getId())
+            .andExpect(status().is2xxSuccessful())
```
Comment (dvoet) [inline service/src/test/java/bio/terra/workspace/app/controller/FolderApiControllerTest.java:327]: Fix unrelated flakey unit test. If deleting the folder is fast enough, the api return 200, not 202 as the test expects. With this change it just cares about success.

File: service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java [hunk 2]
```diff
@@ -522,20 +524,21 @@ public List<WorkspaceDescription> getWorkspaceDescriptions(
...
-                        dbWorkspaceDescription.getWorkspace().workspaceId(),
-                        TpsComponent.WSM,
-                        TpsObjectType.WORKSPACE),
-                "getOrCreatePao");
-        paoMap.put(dbWorkspaceDescription.getWorkspace().workspaceId(), workspacePao);
-      }
+      var workspacePaos =
+          Rethrow.onInterrupted(
+              () ->
+                  tpsApiDispatch.listPaos(
```
Comment (cahrens) [inline service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java:532]: `listPaos` will return an empty list, nut null?

**Review by cahrens — APPROVED** [review]
Assuming the case of no PAOs being returned works.

**Review by yuliadub — APPROVED** [review]

**Review by pshapiro4broad — APPROVED** [review]
LGTM

File: service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java (hunk 2 above)
Comment (dvoet) [reply service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java:532]: correct

**Pull request merged**